src/lib.rs

```diff
@@ -772,6 +775,204 @@ fn pisa_to_ciff_from_paths(
     Ok(())
 }
 
+#[derive(Debug, Deserialize)]
+struct JsonDoc {
+    /// CIFF docid (must be an integer). In a typical CIFF index, this is 0..(num_docs-1).
+    id: i32,
+
+    /// Optional textual content for the document.
+    #[serde(default)]
+    content: String,
+
+    /// A dictionary from token (term) to a score (e.g., frequency). This is optional in the JSON.
+    #[serde(default)]
+    vector: HashMap<String, f64>,
+}
+
+/// PISA to CIFF converter.
+#[derive(Debug, Default, Clone)]
+pub struct JsonlToCiff {
+    input: Option<PathBuf>,
+    output: Option<PathBuf>,
+}
+
+impl JsonlToCiff {
+    /// Set the path of the JSONL file. Required.
+    pub fn input_path<P: Into<PathBuf>>(&mut self, path: P) -> &mut Self {
+        self.input = Some(path.into());
+        self
+    }
+
+    /// Set the output CIFF file path. Required.
+    pub fn output_paths<P: Into<PathBuf>>(&mut self, path: P) -> &mut Self {
+        self.output = Some(path.into());
+        self
+    }
+
+    /// Performs the conversion from JSONL to CIFF.
+    ///
+    /// # Errors
+    ///
+    /// If the input / output is invalid or any I/O / parsing error occurs,
+    /// returns an error.
+    pub fn convert(&self) -> Result<()> {
+        let input_path = self
+            .input
+            .as_ref()
+            .ok_or_else(|| anyhow!("No JSON input path was set"))?;
+        let output_path = self
+            .output
+            .as_ref()
+            .ok_or_else(|| anyhow!("No CIFF output path was set"))?;
+
+        // Open the JSONL file
+        let input_file =
+            File::open(input_path).with_context(|| format!("Cannot open {input_path:?}"))?;
+        let total_input_size = input_file.metadata()?.len();
+
+        let reader = BufReader::new(input_file);
+
+        // We'll store doc-level info:
+        let mut doc_records: Vec<DocRecord> = Vec::new();
+
+        // We'll map "term" -> (docid, tf).
+        // Because CIFF uses integer tf, we round any float to i32.
+        let mut postings_map: HashMap<String, Vec<(i32, i32)>> = HashMap::new();
+
+        let mut total_terms_in_collection: i64 = 0;
+        let mut max_docid: i32 = -1;
+
+        // Read JSON lines
+        eprintln!("Read JSON lines");
+        let pb = ProgressBar::new(total_input_size);
+        pb.set_style(pb_style());
+        for line_result in reader.lines() {
+            let line = line_result?;
+            let jdoc: JsonDoc = match serde_json::from_str(&line) {
```

Comment:
I think instead of `match` we can just do `serde_json::from_str(&line).map_err(|e| Err(anyhow!(...)))?;`